rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java

```diff
@@ -176,6 +176,24 @@ private CachingRlsLbClient(Builder builder) {
     logger.log(ChannelLogLevel.DEBUG, "CachingRlsLbClient created");
   }
 
+  /**
+   * If it is a code that should not be generated by the library, convert the status into one
+   * that is allowed.  Otherwise, return the argument.
+   * @param status status as provided by server
+   * @param args Used for error description
+   * @return Transformed status or argument depending on the Status.code
+   */
+  static Status convertServerStatus(Status status, PickSubchannelArgs args) {
+    String serviceName = args.getMethodDescriptor().getServiceName();
+    String methodName = args.getMethodDescriptor().getBareMethodName();
+    Metadata headers = args.getHeaders();
+
+    return Status.Code.UNAVAILABLE.toStatus().withCause(status.getCause()).withDescription(
+        String.format("Unable to retrieve RLS targets for service=%s, method=%s, headers=%s.  "
```

Comment:
We absolutely must not include all headers, as it could have sensitive fields like oauth tokens. Service and method could be useful, but that's actually the context the caller should have already.

To add useful context, the `RouteLookupRequest` really has the important information. Unfortunately I don't think it is safe to include in the status message because the configuration is unauthenticated; an attacker could trick a server to revealing a sensitive header key (e.g., oauth token). The most we could include is probably just the RLS target name.